BinaryEncoding.md

```diff
@@ -149,18 +176,44 @@ ID: `import`
 The import section declares all imports that will be used in the module.
 
 | Field | Type | Description |
-| ----- |  ----- | ----- |
+| ----- | ---- | ----------- |
 | count | `varuint32` | count of import entries to follow |
 | entries | `import_entry*` | repeated import entries as described below |
 
 #### Import entry
 | Field | Type | Description |
-| ----- |  ----- | ----- |
-| sig_index | `varuint32` | signature index of the import |
+| ----- | ---- | ----------- |
 | module_len | `varuint32` | module string length |
 | module_str | `bytes` | module string of `module_len` bytes |
-| function_len | `varuint32` | function string length |
-| function_str | `bytes` | function string of `function_len` bytes |
+| field_len | `varuint32` | field name length |
+| field_str | `bytes` | field name string of `field_len` bytes |
+| kind | `definition_kind` | the kind of definition being imported |
+
+Followed by, if the `kind` is `Function`:
+
+| Field | Type | Description |
+| ----- | ---- | ----------- |
+| sig_index | `varuint32` | signature index of the import |
+
+or, if the `kind` is `Table`:
+
+| Field | Type | Description |
+| ----- | ---- | ----------- |
+| element_type | `varuint7` | `0x40`, indicating [`anyfunc`](AstSemantics.md#table) |
```

Comment:
So I was actually thinking forward to these cases and my intention was for `element_type` to use the same representation of type as we'd use for local variables when we had GC types.  In particular, IIRC, we'd have the positive "nullary constructors" for primitive types and then some way (prefix or the negative trick) to say "index into the type section" to describe function, struct, and array types.  So given all that, where do we stick "anyfuc"?  My though was that, since these non-nullary constructors can't show up directly, we could use them to mean "any X" (for X being "func", "struct", "array").  Maybe that's too cute; in which case I think "anyfunc" would just be another nullary constructor, but I think it would necessarily need to be disjoint from all the other constructors, right?